### PR TITLE
gift config modal validation

### PR DESCRIPTION
**Pull request by adammeyer**
Simple validation for amount (must be greater than $1)
Add max length of 250 for comments

### CODE REVIEW

File: src/app/productConfig/productConfigModal.tpl.html [hunk 1]
```diff
@@ -43,9 +43,11 @@ <h4 class="panel-title border-bottom-small" translate>
               <label class="btn btn-radio" ng-click="$ctrl.itemConfig.amount = 5000" ng-class="{'active': $ctrl.itemConfig.amount === 5000}"><span class="number">$5000</span></label>
               <label class="btn btn-default-form u-textLeft custom-amount">
                 <div class="form-group form-group-default u-inline">
-                  <input class="form-control form-control-subtle number" min="0" step="any" type="text" ng-model="$ctrl.itemConfig.amount">
+                  <input class="form-control form-control-subtle number" min="1" type="number" name="amount" ng-model="$ctrl.itemConfig.amount" required>
                 </div>
               </label>
+              <span class="help-block" validation="$ctrl.itemConfigForm.amount" show-error="required" translate>You must enter an amount</span>
+              <span class="help-block" validation="$ctrl.itemConfigForm.amount" show-error="min" translate>Amount must be at least $1</span>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Are we ever going to accept non USD currencies? Not a blocker for this PR unless you can think of a generic error message.

File: src/app/productConfig/productConfigModal.tpl.html [hunk 2]
```diff
@@ -125,12 +115,22 @@ <h4 class="panel-title border-bottom-small mt" translate>
           <div class="panel-body pt0">
             <ul class="list-unstyled">
               <li>
-                <a class="commentLink" href="" ng-click="showRecipientComments = !showRecipientComments" translate>Send a Message to {{$ctrl.productData.displayName}}</a>
-                <textarea class="staff-comments give-modal-comments" rows="3" ng-model="$ctrl.itemConfig['recipient-comments']" ng-if="showRecipientComments || $ctrl.itemConfig['recipient-comments']" style="display:block;"></textarea>
+                <a class="commentLink" href="" ng-click="showRecipientComments = !showRecipientComments"><translate>Send a Message to</translate> {{$ctrl.productData.displayName}}</a>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Will the translate directive not work? It seems to [support interpolation](https://angular-gettext.rocketeer.be/dev-guide/annotate/#interpolation).

File: src/app/productConfig/productConfigModal.tpl.html [hunk 3]
```diff
@@ -125,12 +115,22 @@ <h4 class="panel-title border-bottom-small mt" translate>
...
+                <textarea class="staff-comments give-modal-comments"
+                        rows="3"
+                        maxlength="250"
+                        ng-model="$ctrl.itemConfig['recipient-comments']"
+                        ng-if="showRecipientComments || $ctrl.itemConfig['recipient-comments']"
+                        style="display:block;"></textarea>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Does this need `display: block`?

File: src/app/productConfig/productConfigModal.tpl.html (hunk 2 above)
Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: Okay, I'll switch it back.  Thanks.

File: src/app/productConfig/productConfigModal.tpl.html (hunk 3 above)
Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: Yes, the css is hiding it because it used a jquery library to show it.

**Comment by OzzieOrca** [comment]
Besides those minor questions, 👍

File: src/app/productConfig/productConfigModal.tpl.html (hunk 3 above)
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: There's a second spot that has display block too.

Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Ah. Ya I've seen that elsewhere... K leave it

**Pull request merged**